### PR TITLE
Add structs to hold tracepoints with call stacks

### PR DESCRIPTION
**Pull request by Mahmood-Darwish**
This PR adds structs to represent the tracepoints with the callstacks.

Bug: http://b/242823946

### CODE REVIEW

**Review by beckerhe — APPROVED** [review]
This looks good to me but I can't really judge without seeing the usage, so Florian should also approve. He has more context.

File: src/LinuxTracing/PerfEvent.h [hunk 1]
```diff
@@ -259,6 +259,118 @@ struct DmaFenceSignaledPerfEventData {
 };
 using DmaFenceSignaledPerfEvent = TypedPerfEvent<DmaFenceSignaledPerfEventData>;
 
+struct CallchainSchedWakeupPerfEventData {
```
Comment (florian-kuebler) [inline src/LinuxTracing/PerfEvent.h:262]: I suggest to follow the Uprobes naming schemes and go with `SchedWakeupWithCallchainPerfEventData` (same below)

**Review by florian-kuebler — APPROVED** [review]
Besides the naming, LGTM. Thanks

File: src/LinuxTracing/PerfEvent.h [hunk 2]
```diff
@@ -259,6 +259,118 @@ struct DmaFenceSignaledPerfEventData {
...
+  pid_t prev_pid_or_minus_one;
+  pid_t prev_tid;
+  int64_t prev_state;
+  int32_t next_tid;
+  std::unique_ptr<uint64_t[]> regs;
+  uint64_t dyn_size;
+  std::unique_ptr<uint8_t[]> data;
+};
+using StackSchedSwitchPerfEvent = TypedPerfEvent<StackSchedSwitchPerfEventData>;
+
```
[the remark this answers is not in the document]
Comment (Mahmood-Darwish) [reply src/LinuxTracing/PerfEvent.h:373]: Yes, they will be added as a part of another change when I connect the pipeline.

File: src/LinuxTracing/PerfEvent.h (hunk 1 above)
Comment (Mahmood-Darwish) [reply src/LinuxTracing/PerfEvent.h]: Done.

**Pull request merged**